llvm/test/Transforms/TailCallElim/entry-branch-drop-debugloc.ll

```diff
@@ -0,0 +1,57 @@
+; RUN: opt -S -passes=tailcallelim < %s | FileCheck %s
```

Comment:
One more thing, it'd be good to add a comment describing what this is testing for, particularly since the CHECK lines are fairly bare. Related to that, does this test add any coverage that the previous test (`debugloc.ll`) doesn't cover?

Reply:
This test case indeed adds coverage (related to the creation of the recursion accumulator). However, the purpose of the tests is to check that the new entry branch has no debugloc. For this, the previous test is enough. I'll delete the newly added test.